### PR TITLE
feat: Create 'help' command

### PR DESCRIPTION
**Pull request by Daeluse**
Creates a simple 'help' command that will print out commands available given the current context.

Executing 'help' without specifying a known command will show you the top level commands available.

image

Executing 'help' after a command will display the available subcommands.

image

### CODE REVIEW

File: lib/src/commands.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  BosunCmd(this.context)
+      : super(
+            command: 'bosun',
+            description: 'Run a command in a shell',
+            subcommands: [HelpCmd(context)]);
+
+  @override
+  void run(List<String> args, Map<String, dynamic> flags) {
+    // Intentionally left blank
+  }
```
Comment (bradcypert) [inline lib/src/commands.dart:46]: I am confused as to why we need this class.  Whats the purpose of it besides holding the help command?

File: lib/src/commands.dart [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            subcommands: []);
+
+  @override
+  void run(List<String> args, Map<String, dynamic> flags) {
+    String commands = '';
+    for (var element in [...context.subcommands!, ...BosunCmd(context).subcommands!]) {
+      commands += '  ${element.command}\t${element.description ?? ''}\n';
+    }
+    print('''
+A command-line utility for Dart CLI development.
```
Comment (bradcypert) [inline lib/src/commands.dart:19]: Consumers of Bosun should be able to use it without anyone caring that's using Bosun under the hood. Perhaps we add a description on the root Bosun Command and populate this from that field.

File: lib/src/commands.dart [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Usage: <command|dart-file> [arguments]
+
+Global options:
+-h, --help                 Print this usage information.
+
+Available commands:
+$commands
+Run "<command|dart-file> help" for more information about a command.
+See https://github.com/PyreStudios/bosun for detailed documentation.
```
Comment (bradcypert) [inline lib/src/commands.dart:29]: Similar here, we dont necessarily want to say "Hey, this CLI was built with Bosun, check out the docs here." Most of the consumers of that CLI won't really care what its built with. We can probably omit this line.

File: lib/src/commands.dart (hunk 1 above)
Comment (Daeluse) [reply lib/src/commands.dart:46]: This gives us an entry point to define additional global commands offered by the Bosun framework. If you feel that `help` will be the only one of this kind to ever exist, we can safely omit it.

File: lib/src/commands.dart (hunk 2 above)
Comment (Daeluse) [reply lib/src/commands.dart]: Now uses the `description` field of the root command.

File: lib/src/commands.dart (hunk 3 above)
Comment (Daeluse) [reply lib/src/commands.dart]: Omitted this line.

File: lib/src/commands.dart (hunk 1 above)
Comment (bradcypert) [reply lib/src/commands.dart]: I see. This makes more sense to me now. Thoughts on renaming this to something like `BosunInternalRootCommand`?

**Comment by bradcypert** [comment]
One more thought (and Im not married to this one way or another), but should this be `help` `--help` or both?

**Comment by Daeluse** [comment]
> One more thought (and Im not married to this one way or another), but should this be `help` `--help` or both?
I took some inspiration from the Flutter CLI and how it handles help vs --help. And they handle it by supporting `help` from the top level, but --help for any subcommand.

`flutter help`
`flutter run --help`

Does this seem like a good strategy to follow?

File: lib/src/commands.dart (hunk 1 above)
Comment (Daeluse) [reply lib/src/commands.dart]: I think that's a much better name. Do you want Command, or Cmd following the example's naming strategy?

**Comment by bradcypert** [comment]
@Daeluse yeah that sounds great to me.

**Comment by bradcypert** [comment]
@Daeluse any other thoughts or concerns before merging this one in?

**Comment by Daeluse** [comment]
@bradcypert - Nope, I think we're good here.

**Pull request merged**